File: benchmarks/method_call.lua

Toggle = {}

function Toggle:activate ()
	self.state = not self.state
	return self.state
end

function Toggle:new(start_state)
	local o = {state = start_state}
	self.__index =self
	setmetatable(o, self)
	return o
end

function main ()
	local start = os.clock()
	local N = 100000

	local val = 1
  	local toggle = Toggle:new(val)
  	for i=1,N do
		val = toggle:activate()
	end

	print(val and "true" or "false")
	io.write(string.format("elapsed: %.8f\n", os.clock() - start))
end

main()

File: benchmarks/method_call.py
#!/usr/bin/python
from __future__ import print_function

import sys
import time

class Toggle(object):
    def __init__(self, start_state):
        self.bool = start_state
    def activate(self):
        self.bool = not self.bool
        return self.bool

def main():
    start = time.clock()

    NUM = 100000

    val = True
    toggle = Toggle(val)
    for i in range(0,NUM):
        val = toggle.activate()
    if val:
        print("true")
    else:
        print("false")

    print("elapsed: " + str(time.clock() - start))


main()
